conda-installation.md

```diff
@@ -247,3 +269,71 @@ arguments = myscript.py
+
+An important part of improving reproducibility and consistency between runs
+is to ensure that you use the correct/expected versions of your dependencies.
+
+When you run a command like `conda install numpy`, `conda` tries to install
+**the most recent version of `numpy`**. For example, `numpy` version `1.18.2`
+was released on March 17, 2020. To install exactly this version of `numpy`, you
+would run `conda install numpy==1.18.2` (the same works for `pip`). We
+recommend installing with an explicit version to make sure you have exactly 
+the version of a package that you want. This is often called 
+"pinning" or "locking" the version of the package.
+
+If you want a record of what is installed in your environment, or want to 
+reproduce your environment on another computer, conda can create a file, usually
+called `environment.yml`, that describes the exact versions of all of the 
+packages you have installed in an environment.
+This file can be re-used by a different conda command to recreate that 
+exact environment on another computer.
+
+To create an `environment.yml` file from your currently-activated environment, run
+```
+[alice@submit]$ conda env export > environment.yml
+```
+{: .term}
+
+This `environment.yml` will pin the exact version of *every dependency in your
+environment*. This can sometimes be problematic if you are moving between
+platforms because a package version may not be available on some other platform,
+causing an "unsatisfiable dependency" or "inconsistent environment" error.
+A much less strict pinning is
+```
+[alice@submit]$ conda env export --from-history > environment.yml
+```
+{: .term}
+which only lists packages that you installed manually, and **does not pin their
+versions unless you yourself pinned them during installation**. In general,
+exact environment specifications are not guaranteed to be transfer between
+platforms (e.g., between Windows and CentOS). We recommend using the strictest
+possible pinning available to you.
+
+To create an environment from an `environment.yml` file, run
+```
+[alice@submit]$ conda env create -f environment.yml
+```
+{: .term}
+By default, the name of the environment will be whatever the name of the source
+environment was; you can change the name by adding a `-n <name>` option to the
+`conda env create` command.
+
+If you create an `environment.yml`, we recommend checking it into source control
+(like `git`) and making sure to recreate it when you make changes to your environment.
+Putting your environment under source control gives you a way to track how it
+changes along with your own code.
+
+If you are developing software on your local computer for eventual use on 
+the CHTC pool, your workflow might look like this:
+1. Set up a conda environment for local development and install packages as desired.
+2. Once you are ready to run on the CHTC pool, create an `environment.yml` file
+   from your local environment (`conda env export > environment.yml`).
```

Comment:
Should the `--from-history` option be recommended here or not?  I don't have enough experience with it because I'm using to constructing environment files manually.

Reply:
You're far braver than I :)

I think I like the example here not using `--from-history`, since this causes a tighter pinning. We describe when/how to use `--from-history` above, and hopefully if they run into problems they can combine the solutions.

Should we include some details on manually editing environment files? I assume you're doing it to work around unsatisfiable dependency issues while keeping everything else pinned?

Reply:
Sometimes I manually edit environments for unsatisfiable dependency issues, as you said.  Other times it's because the working environment originally included packages that are no longer needed, and I want to iteratively remove packages from the file while checking our code still passes tests.

Manually editing environment files feels slightly out of scope to me for this guide, but I'm ambivalent.  I trust @ChristinaLK's judgement here as someone who has more experience interacting with the target users.

Reply:
I added a brief note about this, but also bolded the "use the strictest possible pinning" suggestion.